### DISCUSSION TITLE
Why do scorers return a different result than process.extract

### DISCUSSION THREAD

**Discussion by maxbachmann** [rapidfuzz/RapidFuzz#165] [Q&A] [answered]
This question was originally asked in https://github.com/maxbachmann/RapidFuzz/discussions/164

> I am also a new user. I'm trying to figure out how to use this library with symbols inside the strings. 
```python
>>> from fuzzywuzzy import fuzz, process
>>> x = ['atk', 'atk%', 'atk!', 'atk$']

>>> process.extract('atk!', x)
[('atk', 100.0, 0), ('atk%', 100.0, 1), ('atk!', 100.0, 2), ('atk$', 100.0, 3)]

>>> fuzz.ratio('atk', 'atk%')
86

>>> process.extract('atk!', x, scorer=fuzz.ratio)
[('atk', 100.0, 0), ('atk%', 100.0, 1), ('atk!', 100.0, 2), ('atk$', 100.0, 3)]
```

@Dosx001 I moved this into a separate discussion, since it is an unrelated issue

**Comment by maxbachmann** [comment, marked as answer]
`process.extract` preprocesses strings by default (see https://maxbachmann.github.io/RapidFuzz/process.html#extract)

> processor (Callable, optional) – Optional callable that reformats the strings. utils.default_process is used by default, which lowercases the strings and trims whitespace

You can disable this behavior by passing `processor=None`:
```python
process.extract('atk!', x, scorer=fuzz.ratio, processor=None)
```
or enable it for scorers by passing `processor=utils.default_process`:
```
fuzz.ratio('atk', 'atk%', processor=utils.default_process)
```

**Comment by Dosx001** [comment]
@maxbachmann Thanks! This was what I needed.

Also at of curiosity why does RapidFuzz and FuzzyWuzzy work like this. Why do I need ```scorer=fuzz.ratio, processor=None```. Why couldn't it work like this out the box?

**Reply by maxbachmann** [reply]
In RapidFuzz I want to stay compatible with FuzzyFuzzy. I selected the same defaults, so most oft the time people only have to change the import. I do not know why SeatGeek selected this default behavior in FuzzyWuzzy.
